groups/ntc/ntcu/ntcu_timestampcorrelator.h

```diff
@@ -0,0 +1,160 @@
+// Copyright 2023 Bloomberg Finance L.P.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef INCLUDED_NTCU_TIMESTAMPCORRELATOR
+#define INCLUDED_NTCU_TIMESTAMPCORRELATOR
+
+#include <bsls_ident.h>
+BSLS_IDENT("$Id: $")
+
+#include <ntccfg_inline.h>
+#include <ntccfg_platform.h>
+#include <ntsa_timestamp.h>
+#include <bdlb_nullablevalue.h>
+#include <bsls_timeinterval.h>
+#include <bslstl_pair.h>
+#include <bsl_cstdint.h>
+
+namespace BloombergLP {
+namespace ntcu {
+
+/// @internal @brief
+/// Correlate transmit timestamps obtained from a socket error queue
+/// with data sent via send() system calls
+///
+/// @details
+/// ntsu::TimestampCorrelator can save and extract timestamps based on their
+/// ids.  Timestamp ids are provided externally
+///
+/// @par Thread Safety
+/// This class is not thread safe.
+///
+/// @ingroup module_ntcu
+class TimestampCorrelator
+{
+  public:
+    typedef bsls::TimeInterval       Timestamp;
+    typedef bsl::uint32_t            ID;
+    typedef bsl::pair<ID, Timestamp> IdTimestamp;
+
+  private:
+    enum { k_RING_BUFFER_SIZE = 64 };
+
+    /// @internal @brief
+    /// RingBuffer is used to store items and extract them based on their id.
+
```

Comment:
Remove the blank line. The documentation style places the comment block immediately proceeding the declaration.

Reply:
Done